### PR TITLE
ENH allow to install all the benchmark in one shot

### PR DESCRIPTION
**Pull request by tomMoral**
Install all deps in one shot in `benchopt install`.

### CODE REVIEW

File: benchopt/utils/dependencies_mixin.py [hunk 1]
```diff
@@ -110,11 +110,52 @@ def install(cls, env_name=None, force=False):
...
+            Name of the conda env where the class should be installed. If
+            None, tries to install it in the current environment.
+        force : boolean (default: False)
+            If set to True, forces reinstallation when using conda.
+
+        Returns
+        -------
+        requirements: list of str
+            List of all requirements for this class.
+        install_shell_scrip: str
```
Comment (agramfort) [inline benchopt/utils/dependencies_mixin.py:133]: script with t

File: benchopt/runner.py [hunk 2]
```diff
@@ -401,10 +401,10 @@ def run_benchmark(benchmark, solver_names=None, forced_solvers=None,
 
     # Load the objective class for this benchmark and the datasets
     objective_class = benchmark.get_benchmark_objective()
-    datasets = benchmark.list_benchmark_datasets()
+    datasets = benchmark.datasets
```
Comment (agramfort) [inline benchopt/runner.py:404]: I would prefer a get_datasets() method as here you have a property that does some compute

File: benchopt/runner.py [hunk 3]
```diff
@@ -401,10 +401,10 @@ def run_benchmark(benchmark, solver_names=None, forced_solvers=None,
...
 
     # Load the solvers and filter them to get the one to run
-    solver_classes = benchmark.list_benchmark_solvers()
+    solver_classes = benchmark.solvers
```
Comment (agramfort) [inline benchopt/runner.py:407]: Get_solvers()

**Review by agramfort — COMMENTED** [review]
Besides lgtm

File: benchopt/runner.py (hunk 2 above)
Comment (tomMoral) [reply benchopt/runner.py:404]: Let's discuss this tomorrow but I think I prefer the property.

**Review by agramfort — APPROVED** [review]

**Review by agramfort — APPROVED** [review]
Merge if happy @tomMoral

**Pull request merged**